image-patching/stopgap-script/nuage_overcloud_full_patch.py

```diff
@@ -406,33 +406,45 @@ def image_patching(nuage_config):
 
 def check_config(nuage_config):
     missing_config = []
-    for key in ["ImageName", "DeploymentType", "RepoFile", "VRSRepoNames"]:
+    for key in ["ImageName", "RepoFile", "VRSRepoNames"]:
         if not (nuage_config.get(key)):
             missing_config.append(key)
     if missing_config:
         logger.error("Please provide missing config %s value "
                      "in your config file. \n" % missing_config)
         sys.exit(1)
-    if "avrs" in nuage_config["DeploymentType"] and "ovrs" in nuage_config["DeploymentType"]:
+    file_exists(nuage_config["ImageName"])
+
+    if not nuage_config["DeploymentType"]:
+        logger.info("Overcloud Image will be patched with Nuage VRS rpms")
+    elif "avrs" in nuage_config["DeploymentType"] and "ovrs" in nuage_config["DeploymentType"]:
         logger.error(
             "Currently Nuage doesn't support both AVRS and OVRS deployment together"
             "Please choose only one between them")
         sys.exit(1)
-    if "avrs" in nuage_config["DeploymentType"]:
+    elif nuage_config["DeploymentType"] is "avrs":
+        logger.info("Overcloud Image will be patched with Nuage VRS & AVRS rpms")
         if not nuage_config.get("AVRSRepoNames"):
             logger.error("Please provide AVRSRepoNames for AVRS deployment")
             sys.exit(1)
-    if "ovrs" in nuage_config["DeploymentType"]:
+    elif nuage_config["DeploymentType"] is "ovrs":
+        logger.info("Overcloud Image will be patched with OVRS rpms")
         for reponame in ["KernelRepoNames", "MellanoxRepoNames"]:
             if not nuage_config.get(reponame):
                 logger.error(
                     "Please provide %s for OVRS deployment" % reponame)
                 sys.exit(1)
-
+    else:
+        logger.error(
+            "DeploymentType config option was not correctly entered"
```

Comment:
lets say "DeploymentType config option \<option user provided\> is not valid"

Reply:
Done